### PR TITLE
1. improve junit-mock for testing

### PR DESCRIPTION
**Pull request by dfrankland**
improves junit-mocking for better tests, especially ones concerning validation

### CODE REVIEW

File: junit-mock/src/lib.rs [hunk 1]
```diff
@@ -69,15 +70,30 @@ pub struct Options {
...
+impl Default for Options {
+    fn default() -> Self {
+        Options::try_parse_from([""]).unwrap()
+    }
+}
+
+#[test]
+fn options_can_be_defaulted_without_panicing() {
+    Options::default();
+}
```
Comment (dfrankland) [inline junit-mock/src/lib.rs:82]: Removes the need for `unwrap`ing anytime we need to create options in tests or elsewhere

File: junit-mock/src/lib.rs [hunk 2]
```diff
@@ -69,15 +70,30 @@ pub struct Options {
...
+#[derive(Debug, Parser, Clone)]
 #[group()]
 pub struct GlobalOptions {
     /// Seed for all generated data, defaults to randomly generated seed
     #[arg(long)]
     pub seed: Option<u64>,
+
+    /// Timestamp for all data to be based on, defaults to now
+    #[arg(long)]
+    pub timestamp: Option<DateTime<FixedOffset>>,
```
Comment (dfrankland) [inline junit-mock/src/lib.rs:93]: This allows creating more deterministic mocks which is great for validation checking

File: junit-mock/src/lib.rs [hunk 3]
```diff
@@ -173,26 +189,40 @@ pub struct TestRerunOptions {
...
+    #[arg(long, value_parser = clap::value_parser!(u8).range(0..=100), default_value = "50")]
+    pub test_rerun_sys_out_percentage: u8,
+
+    /// Inclusive range of time between test case timestamps
+    #[arg(long, num_args = 1..=2, value_names = ["DURATION_RANGE_START", "DURATION_RANGE_END"], default_values = ["30s", "1m"])]
+    pub test_rerun_duration_range: Vec<humantime::Duration>,
+
+    /// The chance of a system error message being added to the test rerun
+    #[arg(long, value_parser = clap::value_parser!(u8).range(0..=100), default_value = "50")]
+    pub test_rerun_sys_err_percentage: u8,
```
Comment (dfrankland) [inline junit-mock/src/lib.rs:203]: Added some extra options since we want to try parsing these later on in tests

File: junit-mock/src/lib.rs [hunk 4]
```diff
@@ -253,13 +287,17 @@ impl JunitMock {
...
+        reports.as_ref().iter().enumerate().try_fold(
+            Vec::new(),
+            |mut acc, (i, report)| -> Result<Vec<PathBuf>> {
+                let path = directory.as_ref().join(format!("junit-{}.xml", i));
+                let file = File::create(&path)?;
+                report.serialize(file)?;
+                acc.push(path);
+                Ok(acc)
+            },
+        )
```
Comment (dfrankland) [inline junit-mock/src/lib.rs:300]: makes it easier to iterate over generated files instead of hard-coding paths in tests

**Review by gnalh — APPROVED** [review]

**Pull request merged**